specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-06-02-preview/fleets.json

```diff
@@ -0,0 +1,741 @@
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Deletes a fleet member resource.": {
+            "$ref": "./examples/FleetMembers_Delete.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerService/fleets/{fleetName}/members": {
+      "get": {
+        "tags": [
+          "FleetMembers"
+        ],
+        "operationId": "FleetMembers_ListByFleet",
+        "summary": "Lists the members of a fleet.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/FleetNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/FleetMembersListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Lists the members of a fleet.": {
+            "$ref": "./examples/FleetMembers_List.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Fleet": {
+      "type": "object",
+      "properties": {
+        "etag": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Resource Etag."
+        },
+        "properties": {
+          "description": "Properties of a fleet.",
+          "$ref": "#/definitions/FleetProperties",
+          "x-ms-client-flatten": true
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "description": "The fleet resource which contains multiple Kubernetes clusters as its members."
+    },
+    "FleetProperties": {
+      "type": "object",
+      "description": "Properties of a fleet.",
+      "properties": {
+        "hubProfile": {
+          "description": "The FleetHubProfile configures the fleet's hub. it is enabled by default and can be disabled by setting the hubMode property.",
+          "$ref": "#/definitions/FleetHubProfile",
+          "x-ms-client-flatten": true
```

Comment:
hm, it actually doesn't flatten?
see the resulting SDK from this PR:

https://github.com/azure-sdk/azure-sdk-for-go/blob/bdf79d2f2ed08db61922ba8db0b2d72bf4483348/sdk/resourcemanager/containerservice/armcontainerservice/models.go#L470

Reply:
flatten:

```
Fleet : {
   hubProfile: {
    ...
    }
}
```

if you don't, you end up with : 

```
Fleet : {
    hubProfile: {
        FleetHubProfile: {
            ...
        }
    }
}
```

that seems wrong?
same as for properties, we use flatten to get :
```
Fleet: {
   properties {
      ...
   }
```

instead of : 
```
Fleet: {
    properties: {
         FleetProperties {
          ...
         }
    }
}
```